staging/src/k8s.io/apiserver/pkg/server/options/admission.go

```diff
@@ -129,7 +129,23 @@ func (a *AdmissionOptions) ApplyTo(
 }
 
 func (a *AdmissionOptions) Validate() []error {
+	if a == nil {
+		return nil
+	}
+
 	errs := []error{}
+
+	registeredPluginsMap := make(map[string]bool)
+	registeredPlugins := a.Plugins.Registered()
+	for _, name := range registeredPlugins {
+		registeredPluginsMap[name] = true
+	}
+
+	for _, name := range a.PluginNames {
+		if !registeredPluginsMap[name] {
+			errs = append(errs, fmt.Errorf("--admission-control plugin %q is invalid", name))
```

Comment:
this is not necessarily configured through the flag. Better just call it `admission plugin`.

Reply:
make sense

Reply:
a.PluginNames here is all through --admission-control flag.  Validate called before ApplyTo.

Reply:
Downstream projects might set up the options without flags and run the validation on them, e.g. OpenShift.

Reply:
Understand.